### PR TITLE
refactor: reduce CDeterministicMNManager globals use

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Dependent on https://github.com/dashpay/dash/pull/5929

## Breaking Changes

None. Changes are limited to refactoring, no logical changes have been made.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/validationinterface.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include <primitives/block.h>
 #include <primitives/transaction.h>
 #include <scheduler.h>
-
+#include <evo/deterministicmns.h>
```
Comment (knst) [inline src/validationinterface.cpp:15]: you don't need this `<include>` here probably, seems as just forward declaration is enough

File: src/rpc/governance.cpp [hunk 2]
```diff
@@ -1005,7 +1005,9 @@ static UniValue voteraw(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Malformed base64 encoding");
     }
 
-    auto dmn = node.dmnman->GetListAtChainTip().GetValidMNByCollateral(outpoint);
+    CHECK_NONFATAL(node.dmnman);
```
Comment (knst) [inline src/rpc/governance.cpp:1008]: maybe better to implement `EnsureMNMan` same as `EnsureMempool` and `EnsureLLMQContext` ?

File: src/validationinterface.cpp (hunk 1 above)
Comment (kwvg) [reply src/validationinterface.cpp:15]:
```
validationinterface.cpp:291:25: error: by-copy capture of variable 'tip_mn_list' with incomplete type 'const CDeterministicMNList'
    auto event = [vote, tip_mn_list, this] {
                        ^
./validationinterface.h:23:7: note: forward declaration of 'CDeterministicMNList'
class CDeterministicMNList;
      ^
```

File: src/rpc/governance.cpp (hunk 2 above)
Comment (kwvg) [reply src/rpc/governance.cpp:1008]: Don't know if we should implement `Ensure`(`Any`)`*` unless it's significant (like `LLMQContext`)

File: src/governance/object.h [hunk 3]
```diff
@@ -223,16 +224,16 @@ class CGovernanceObject
 
     // CORE OBJECT FUNCTIONS
 
-    bool IsValidLocally(std::string& strError, bool fCheckCollateral) const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+    bool IsValidLocally(const CDeterministicMNList& tip_mn_list, std::string& strError, bool fCheckCollateral) const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```
Comment (knst) [inline src/governance/object.h:227]: consider create a new class such as `CGovernanceObjectVotable` which is:

```cpp
class CGovernanceObjectVotable
{
public:
    CGovernanceObject m_obj;
    const CDeterministicMNList tip_mn_list;
  CGovernanceObjectVotable(const CDeterministicMNList& mn_list, CGovernanceObject m_obj);
....
....

};
```

And move all methods that requires CDeterministicMNList such as IsValidLocally, UpdateLocalValidy, etc - to this new class.

In this case code below such as rpc implementation can be rewritten like that:
```
 const CGovernanceObjectVotable* pGovObj = Governance::MakeVotable(node.govman->FindConstGovernanceObject(hash), node.dmnman->GetListAtChainTip());

    UniValue objFundingResult(UniValue::VOBJ);
    objFundingResult.pushKV("AbsoluteYesCount",  pGovObj->GetAbsoluteYesCount(VOTE_SIGNAL_FUNDING));
    objFundingResult.pushKV("YesCount",  pGovObj->GetYesCount(VOTE_SIGNAL_FUNDING));
    objFundingResult.pushKV("NoCount",  pGovObj->GetNoCount(VOTE_SIGNAL_FUNDING));
    objFundingResult.pushKV("AbstainCount",  pGovObj->GetAbstainCount(VOTE_SIGNAL_FUNDING));
```

File: src/validationinterface.cpp (hunk 1 above)
Comment (knst) [reply src/validationinterface.cpp:15]: can you capture here by reference instead?

**Review by PastaPastaPasta — DISMISSED** [review]
utACK f8d18539038230d493403016d30b431742faa305

File: src/governance/object.h (hunk 3 above)
Comment (PastaPastaPasta) [reply src/governance/object.h:227]: I don't think this is needed in this PR; maybe a future PR

File: src/validationinterface.cpp (hunk 1 above)
Comment (knst) [reply src/validationinterface.cpp:15]: ping: 1e0bb3b7ea0022e308beb50bf01f384731d27de3

**Review by knst — COMMENTED** [review]
please remove this freshly introduced circular dependency: 1e0bb3b7ea0022e308beb50bf01f384731d27de3

Or create follow-up PR for it.

otherwise LGTM.

**Review by PastaPastaPasta — APPROVED** [review]
(self) utACK fa6847d2fb91125fa1354c400a60784f9136ebae

**Review by knst — APPROVED** [review]
utACK fa6847d2fb91125fa1354c400a60784f9136ebae

**Pull request merged**